src/ncp/ncp_base.cpp

```diff
@@ -3494,13 +3544,53 @@ ThreadError NcpBase::SetPropertyHandler_MAC_SCAN_STATE(uint8_t header, spinel_pr
             break;
 
         case SPINEL_SCAN_STATE_ENERGY:
-            errorCode = otEnergyScan(
-                mInstance,
-                mChannelMask,
-                mScanPeriod,
-                &HandleEnergyScanResult_Jump,
-                this
-            );
+#if OPENTHREAD_ENABLE_RAW_LINK_API
+            if (otLinkRawIsEnabled(mInstance))
+            {
+                if (mCurScanChannel == NCP_INVALID_SCAN_CHANNEL)
+                {
+                    uint8_t scanChannel;
+                    uint16_t scanDuration;
+
+                    parsedLength = spinel_datatype_unpack(
```

Comment:
I understand your reasons (specially the first one). However, Spinel is a property based model. If you look at how a high-level operation like "forming"/"join"ing a network (from wpantund) works, it maps to a sequence of spinel property assignments. 

I still think we should have consistent format for value of each property (i.e., how a property is encoded should remain the same and independent of its value...)

I see the following approaches to proceed: 

1. Defining and using  a new  spinel property for raw energy scan which include all parameters in its definition.
2. Using multiple properties for the raw APIs (same as current approach for scan)
3. Changing the logic for the current non-raw case to include the channel mask and duration as part of a single property set...(this requires a change in wpantund...)

My preference/vote would be for 1...

@darconeous  mind providing your inputs about this... Thanks..